static/js/message_list_data.js

```diff
@@ -37,6 +37,47 @@ MessageListData.prototype = {
         return this._items[this._items.length - 1];
     },
 
+    select_idx: function () {
+        if (this._selected_id === -1) {
+            return;
+        }
+        var ids = _.pluck(this._items, 'id');
+
+        var i = ids.indexOf(this._selected_id);
+        if (i === -1) {
+            return;
+        }
+        return i;
+    },
```

Comment:
It seems like we should be able to do this without doing a linear scan.  If we didn't have the `pluck` part, we could do `_.indexOf(ids, this._selected_id, true)` to do a binary search.

Given that we do have the `ids` field to extract, we'd need to do something messier (probably starting with switching from `underscore.js` to its newer fork `lodash.js`, so that we can just implement this as `_.sortedIndexBy(this._items, this._selected_id, function(item) { return item.id; })`.